exporter/awskinesisexporter/internal/batch/batch.go

```diff
@@ -78,7 +75,13 @@ func New(opts ...Option) *Batch {
 }
 
 func (b *Batch) AddRecord(raw []byte, key string) error {
-	record, err := b.compression.Do(raw)
+
+	compressor, err := compress.NewCompressor(b.compressionType)
```

Comment:
I think the source of the race condition is here that `compression.Do` holds the buffer as an instance and so if you call it in parallel, it breaks. 

You've solved this by creating a new compressor each time we want to add a record. I think it's probably better that we just convert these compressors to: 

```
type compressor func(in []byte) []byte
```

So they don't hold any internal state and we don't need to create a new compressor struct every time.

Reply:
sounds great, I'll do it in this way